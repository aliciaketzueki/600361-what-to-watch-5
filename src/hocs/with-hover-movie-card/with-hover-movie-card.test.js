import React, {Fragment} from "react";
import renderer from "react-test-renderer";
import withHoverMovieCard from "./with-hover-movie-card";
import PropTypes from "prop-types";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withHoverMovieCard(MockComponent);

describe(`Render withHoverMovieCard`, () => {
  it(`Should withHoverMovieCard render correctly`, () => {
    const tree = renderer.create(
        <MockComponentWrapped>
          <Fragment />
        </MockComponentWrapped>, {
          createNodeMock() {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
