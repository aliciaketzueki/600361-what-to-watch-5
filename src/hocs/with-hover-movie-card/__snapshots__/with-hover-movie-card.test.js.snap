// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render withHoverMovieCard Should withHoverMovieCard render correctly 1`] = `
<div>
  <video />
</div>
`;
